db/database_test.go

```diff
@@ -338,13 +338,279 @@ func TestGetRemovedAsUser(t *testing.T) {
+
+	// Get the deleted doc with its history; equivalent to GET with ?revs=true, while still resident in the rev cache
+	body, err := db.Get1xRevBody("doc1", rev2ID, true, nil)
+	require.NoError(t, err, "Error getting 1x rev body")
+
+	rev1GenID := rev1ID[2:]
+	rev2GenID := rev2ID[2:]
+	bodyExpected := Body{
+		"k2":       "v2",
+		"channels": []string{"ABC"},
+		BodyRevisions: Revisions{
+			RevisionsStart: 2,
+			RevisionsIds:   []string{rev2GenID, rev1GenID},
+		},
+		BodyId:  "doc1",
+		BodyRev: rev2ID,
+	}
+	require.Equal(t, bodyExpected, body)
+
+	// Flush the revision cache and purge the old revision backup.
+	db.FlushRevisionCacheForTest()
+	err = db.PurgeOldRevisionJSON("doc1", rev2ID)
+	require.NoError(t, err, "Error purging old revision JSON")
+
+	// Generate delta between rev2ID and rev3ID (toRevision "rev2ID" is channel removal)
+	// as a user who doesn't access to the removed revision via any another channel.
+	authenticator := auth.NewAuthenticator(db.Bucket, db)
+	user, err := authenticator.NewUser("alice", "pass", base.SetOf("NBC"))
+	require.NoError(t, err, "Error creating user")
+
+	db.user = user
+	db.DbStats.InitDeltaSyncStats()
+
+	delta, redactedRev, err := db.GetDelta("doc1", rev2ID, rev3ID)
+	require.Equal(t, base.HTTPErrorf(404, "missing"), err)
+	assert.Nil(t, delta)
+	assert.Nil(t, redactedRev)
+
+	// Generate delta between rev2ID and rev3ID (toRevision "rev2ID" is channel removal)
+	// as a user who has access to the removed revision via another channel.
+	user, err = authenticator.NewUser("bob", "pass", base.SetOf("ABC"))
+
+	db.user = user
+	db.DbStats.InitDeltaSyncStats()
+
+	delta, redactedRev, err = db.GetDelta("doc1", rev2ID, rev3ID)
+	require.Equal(t, base.HTTPErrorf(404, "missing"), err)
+	assert.Nil(t, delta)
+	assert.Nil(t, redactedRev)
+}
+
+// Test delta sync behavior when the toRevision is a channel removal.
+func TestDeltaSyncWhenToRevIsChannelRemoval(t *testing.T) {
+	db := setupTestDB(t)
+	defer db.Close()
+
+	// Create the first revision of doc1.
+	rev1Body := Body{
+		"k1":       "v1",
+		"channels": append([]string{"ABC", "NBC"}),
+	}
+	rev1ID, _, err := db.Put("doc1", rev1Body)
+	require.NoError(t, err, "Error creating doc")
+
+	// Create the second revision of doc1 on channel ABC as removal from channel NBC.
+	rev2Body := Body{
+		"k2":       "v2",
+		"channels": []string{"ABC"},
+		BodyRev:    rev1ID,
+	}
+	rev2ID, _, err := db.Put("doc1", rev2Body)
+	require.NoError(t, err, "Error creating doc")
+
+	// Create the third revision of doc1 on channel ABC.
+	rev3Body := Body{
+		"k3":       "v3",
+		"channels": []string{"ABC"},
+		BodyRev:    rev2ID,
+	}
+	rev3ID, _, err := db.Put("doc1", rev3Body)
+	require.NoError(t, err, "Error creating doc")
+	require.NotEmpty(t, rev3ID, "Error creating doc")
+
+	// Get the deleted doc with its history; equivalent to GET with ?revs=true, while still resident in the rev cache
+	body, err := db.Get1xRevBody("doc1", rev2ID, true, nil)
```

Comment:
Similar question to above, is this Get required?

Reply:
This GET is not necessary in the context of this test like the above one. I thought of including it initially since the subsequent step is to remove the revision from cache and body backup. But I see what you mean and we can remove this too to make the test more readable.